packages/region/src/services/region-module.ts

```diff
@@ -0,0 +1,230 @@
+
+  constructor(
+    {
+      baseRepository,
+      regionService,
+      countryService,
+      currencyService,
+    }: InjectedDependencies,
+    protected readonly moduleDeclaration: InternalModuleDeclaration
+  ) {
+    // @ts-ignore
+    super(...arguments)
+    this.baseRepository_ = baseRepository
+    this.regionService_ = regionService
+    this.countryService_ = countryService
+    this.currencyService_ = currencyService
+  }
+
+  __hooks = {
+    onApplicationStart: async () => await this.createCountriesAndCurrencies(),
+  }
+
+  __joinerConfig(): ModuleJoinerConfig {
+    return joinerConfig
+  }
+
+  async create(
+    data: CreateRegionDTO[],
+    sharedContext?: Context
+  ): Promise<RegionDTO[]>
+  async create(
+    data: CreateRegionDTO,
+    sharedContext?: Context
+  ): Promise<RegionDTO>
+  @InjectManager("baseRepository_")
+  async create(
+    data: CreateRegionDTO | CreateRegionDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<RegionDTO | RegionDTO[]> {
+    const input = Array.isArray(data) ? data : [data]
+
+    const result = await this.create_(input, sharedContext)
+
+    return await this.baseRepository_.serialize<RegionDTO[]>(
+      Array.isArray(data) ? result : result[0],
+      {
+        populate: true,
+      }
+    )
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  async create_(
+    data: CreateRegionDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<Region[]> {
+    let currencies = await this.currencyService_.list(
+      { code: data.map((d) => d.currency_code.toLowerCase()) },
+      {},
+      sharedContext
+    )
+
+    let currencyMap = new Map(currencies.map((c) => [c.code.toLowerCase(), c]))
+    for (const reg of data) {
+      const lowerCasedCurrency = reg.currency_code.toLowerCase()
+      if (!currencyMap.has(lowerCasedCurrency)) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          `Currency with code: ${reg.currency_code} was not found`
+        )
+      }
+
+      reg.currency = currencyMap.get(lowerCasedCurrency) as RegionCurrencyDTO
+    }
+
+    const result = await this.regionService_.create(data, sharedContext)
+
+    return result
+  }
+
+  async update(
+    data: UpdateRegionDTO[],
+    sharedContext?: Context
+  ): Promise<RegionDTO[]>
+  async update(
+    data: UpdateRegionDTO,
+    sharedContext?: Context
+  ): Promise<RegionDTO>
+  @InjectTransactionManager("baseRepository_")
+  async update(
+    data: UpdateRegionDTO | UpdateRegionDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<RegionDTO | RegionDTO[]> {
+    const input = Array.isArray(data) ? data : [data]
+
+    const result = await this.regionService_.update(input, sharedContext)
+
+    return await this.baseRepository_.serialize<RegionDTO[]>(
+      Array.isArray(data) ? result : result[0],
+      {
+        populate: true,
+      }
```

Comment:
suggestion: same as above, phil open a pr for the update, I forgot in the base class